include/xwayland/selection.h

```diff
@@ -28,19 +28,34 @@ struct wlr_xwm_selection_transfer {
 	// when receiving from x11
 	int property_start;
 	xcb_get_property_reply_t *property_reply;
+
+	// FIXME: these are only used for X11 -> Wayland for now, but are initialized
+	// for both directions.
+	//
+	// Wayland -> X11 continues to use the `wlr_xwm_selection` versions.
+	xcb_window_t window;
+	xcb_timestamp_t timestamp;
+
+	struct wl_list incoming_link;
```

Comment:
We could use a single `struct wl_list link` field which is either outgoing, either incoming. That way `xwm_selection_transfer_finish` can just `wl_list_remove(&transfer->link)`.

Reply:
Good idea, done.